packages/wasm-dpp/src/dash_platform_protocol.rs

```diff
@@ -57,188 +28,168 @@ impl DashPlatformProtocol {
-        self.data_contract.clone()
-    }
-
-    #[wasm_bindgen(getter=document)]
-    pub fn document(&self) -> DocumentFacadeWasm {
-        self.document.clone()
+        let dpp = DashPlatformProtocol::new(protocol_version);
+        Ok(DashPlatformProtocolWasm(dpp))
     }
 
+    // #[wasm_bindgen(getter = dataContract)]
+    // pub fn data_contract(&self) -> DataContractFacadeWasm {
+    //     self.data_contract.clone()
+    // }
+    //
+    // #[wasm_bindgen(getter=document)]
+    // pub fn document(&self) -> DocumentFacadeWasm {
+    //     self.document.clone()
+    // }
+    //
     #[wasm_bindgen(getter = identity)]
     pub fn identity(&self) -> IdentityFacadeWasm {
-        self.identity.clone()
-    }
-
-    #[wasm_bindgen(getter = stateTransition)]
-    pub fn state_transition(&self) -> StateTransitionFacadeWasm {
-        self.state_transition.clone()
-    }
-
-    #[wasm_bindgen(getter = protocolVersion)]
-    pub fn protocol_version(&self) -> u32 {
-        self.protocol_version
-    }
-
-    #[wasm_bindgen(js_name = getProtocolVersion)]
-    pub fn get_protocol_version(&self) -> u32 {
-        self.protocol_version()
-    }
-
-    #[wasm_bindgen(js_name = setProtocolVersion)]
-    pub fn set_protocol_version(&mut self, protocol_version: u32) -> Result<(), JsValue> {
-        self.init(
-            protocol_version,
-            self.state_repository.clone(),
-            self.bls.clone(),
-            self.entropy_generator.clone(),
-        )
-    }
-
-    #[wasm_bindgen(js_name = setStateRepository)]
-    pub fn set_state_repository(
-        &mut self,
-        state_repository: ExternalStateRepositoryLike,
-    ) -> Result<(), JsValue> {
-        self.init(
-            self.protocol_version,
-            state_repository,
-            self.bls.clone(),
-            self.entropy_generator.clone(),
-        )
-    }
-
-    #[wasm_bindgen(js_name = getStateRepository)]
-    pub fn get_state_repository(&self) -> ExternalStateRepositoryLike {
-        self.state_repository.clone()
-    }
-
-    fn init(
-        &mut self,
-        protocol_version: u32,
-        state_repository: ExternalStateRepositoryLike,
-        bls_adapter: BlsAdapter,
-        entropy_generator: ExternalEntropyGenerator,
-    ) -> Result<(), JsValue> {
-        let (identity_facade, document_facade, data_contract_facade, state_transition_facade) =
-            create_facades(
-                self.public_keys_validator.clone(),
-                protocol_version,
-                state_repository.clone(),
-                bls_adapter,
-                entropy_generator,
-            )?;
-
-        self.protocol_version = protocol_version;
-        self.identity = identity_facade;
-        self.document = document_facade;
-        self.data_contract = data_contract_facade;
-        self.state_transition = state_transition_facade;
-        self.state_repository = state_repository;
-
-        Ok(())
+        self.0.identities().into()
```

Comment:
We create an instance of facade every call to `dpp.identities`?

Reply:
Yes, we can not return a reference. Previously we were cloning on every call. 
Not sure there's a big difference